src/rpdk/core/boto_helpers.py

```diff
@@ -32,19 +32,32 @@ def _known_error(msg):
     return session
 
 
-def get_temporary_credentials(session, key_names=BOTO_CRED_KEYS, role_arn=None):
+def get_temporary_credentials(
+    session, key_names=BOTO_CRED_KEYS, role_arn=None, headers=None
+):
     sts_client = session.client(
         "sts",
         endpoint_url=get_service_endpoint("sts", session.region_name),
         region_name=session.region_name,
     )
+
+    if headers and len(headers) == 2:
+        # Inject headers through the event system.
+        def inject_header(params):
+            params["headers"]["x-amz-source-account"] = headers[0]
+            params["headers"]["x-amz-source-arn"] = headers[1]
+
```

Comment:
Wouldn't using a dictionary be preferable here to increase readability?  e.g. headers["source-account"] and headers["source-arn"]